src/pages/api/search.ts

```diff
@@ -0,0 +1,75 @@
+import { createApiFetch } from 'utils/apiFetch';
+import makeSearchRequest from 'api/utils/makeSearchRequest';
+import { NextApiRequest, NextApiResponse } from 'next';
+
+import { SEARCH_DATA_TYPE } from 'types/search';
+
+/**
+ * To use
+ *
+ * fetch('/api/search?orgId=1', {
+ *   method: "post",
+ *   body: {q: "some search string"}
+ * })
+ */
+
+const search = async (
+  req: NextApiRequest,
+  res: NextApiResponse
+): Promise<void> => {
+  const {
+    method,
+    query: { orgId },
+    body: { q },
+  } = req;
+
+  // Return error if method other than POST
+  if (method !== 'POST') {
+    return res
+      .setHeader('Allow', ['POST'])
+      .status(405)
+      .json({ error: `Method ${method} Not Allowed` });
+  }
+
+  // Validate orgId, return error if not valid
+  if (!orgId) {
+    return res
+      .status(400)
+      .json({ error: 'Query Parameter "orgId" not provided' });
+  }
+  if (typeof orgId === 'object') {
+    return res
+      .status(400)
+      .json({ error: 'Query Parameter "orgId" must by a single value' });
+  }
+
+  const query = { orgId, q };
+  const apiFetch = createApiFetch(req.headers);
+
+  try {
+    const peopleRequest = async () => {
+      return makeSearchRequest(SEARCH_DATA_TYPE.PERSON, query, apiFetch);
+    };
+    const campaignRequest = async () => {
+      return makeSearchRequest(SEARCH_DATA_TYPE.CAMPAIGN, query, apiFetch);
+    };
+    const taskRequest = async () => {
+      return makeSearchRequest(SEARCH_DATA_TYPE.TASK, query, apiFetch);
+    };
+
+    const results = await Promise.all([
+      peopleRequest(),
+      campaignRequest(),
+      taskRequest(),
```

Comment:
I remember looking at this together, but I can't remember why we have these wrapper functions. Why not just use `makeSearchRequest(SEARCH_DATA_TYPE.PERSON, query, apiFetch)` here?

Reply:
ah i did it that way because it's easier to read for me, but i can change it